File: src/main/java/no/nav/bidrag/dokument/aop/AspectExceptionLogger.java
package no.nav.bidrag.dokument.aop;

import no.nav.bidrag.commons.ExceptionLogger;
import org.aspectj.lang.JoinPoint;
import org.aspectj.lang.annotation.AfterThrowing;
import org.aspectj.lang.annotation.Aspect;
import org.springframework.stereotype.Component;

@Aspect
@Component
public class AspectExceptionLogger {

  private final ExceptionLogger exceptionLogger;

  public AspectExceptionLogger(ExceptionLogger exceptionLogger) {
    this.exceptionLogger = exceptionLogger;
  }

  @AfterThrowing(pointcut = "within (no.nav.bidrag.dokument.controller..*)", throwing = "exception")
  public void logException(JoinPoint joinPoint, Exception exception) {
    exceptionLogger.logException(exception, String.valueOf(joinPoint.getSourceLocation().getWithinType()));
  }
}

File: src/main/java/no/nav/bidrag/dokument/aop/AspectExceptionLogger.kt
package no.nav.bidrag.dokument.aop

import mu.two.KotlinLogging
import no.nav.bidrag.dokument.sikkerLogg
import org.aspectj.lang.JoinPoint
import org.aspectj.lang.annotation.AfterThrowing
import org.aspectj.lang.annotation.Aspect
import org.springframework.stereotype.Component

private val log = KotlinLogging.logger {}

@Aspect
@Component
class AspectExceptionLogger {
    @AfterThrowing(pointcut = "within (no.nav.bidrag.dokument.controller..*)", throwing = "exception")
    fun logException(joinPoint: JoinPoint, exception: Exception?) {
        log.warn(exception) { "Det skjedde en feil i controller metoden ${joinPoint.signature.toShortString()}. Se sikkerlogg for detaljer" }
        sikkerLogg.warn(exception) { "Det skjedde en feil i controller metoden ${joinPoint.signature.toShortString()} | Argumenter => ${joinPoint.args.joinToString(", ")}" }
    }
}
